### PR TITLE
feat: support b2 store

### PR DESCRIPTION
**Pull request by irrun**
### Description

feat: support [b2](https://www.backblaze.com/b2/cloud-storage.html) as store

### Rationale

for sp store extension

### CODE REVIEW

File: store/piecestore/storage/b2_test.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/stretchr/testify/assert"
+)
+
+func setupB2Test(t *testing.T) *b2Store {
+	// "https://s3.us-east-005.backblazeb2.com/greenfieldsp"
```
Comment (sysvm) [inline store/piecestore/storage/b2_test.go:16]: don't add real b2 bucketURL in code.

File: store/piecestore/storage/b2_test.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+func TestB2_CreateSuccess(t *testing.T) {
+	store := setupB2Test(t)
+	cases := []struct {
+		name      string
+		req       s3.CreateBucketInput
+		resp      s3.CreateBucketOutput
+		wantedErr error
+	}{
+		{
+			name:      "1",
```
Comment (sysvm) [inline store/piecestore/storage/b2_test.go:29]: the name may use a complete name to describe its function.

File: store/piecestore/storage/b2_test.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+	cases := []struct {
+		name      string
+		key       string
+		offset    int64
+		limit     int64
+		resp      s3.GetObjectOutput
+		wantedErr error
+	}{
+		{
+			name:      "1",
```
Comment (sysvm) [inline store/piecestore/storage/b2_test.go:68]: ditto

File: test/e2e/piecestore/b2_test.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/bnb-chain/greenfield-storage-provider/store/piecestore/storage"
+)
+
+const (
+	b2BucketURL = "https://s3.us-east-005.backblazeb2.com/greenfieldsp"
```
Comment (sysvm) [inline test/e2e/piecestore/b2_test.go:12]: don't add real b2 bucketURL in code.

File: store/piecestore/storage/b2.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+		isVirtualHostStyle = false
+	} else {
+		// Virtual hosted style: https://<bucketName>.s3.<region>.backblazeb2.com(.cn)
+		bucketName = strings.SplitN(endpoint, ".s3", 2)[0]
+		endpoint = endpoint[len(bucketName)+1:]
+		isVirtualHostStyle = true
+	}
+
+	region = parseRegion(endpoint)
+	return
```
Comment (sysvm) [inline store/piecestore/storage/b2.go:103]: Returns the result explicitly.

File: store/piecestore/storage/b2.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	s3Store
+}
+
+func newB2Store(cfg ObjectStorageConfig) (ObjectStorage, error) {
+	b2Session, bucket, err := b2SessionCache.newB2Session(cfg)
+	if err != nil {
+		log.Errorw("failed to new b2 session", "error", err)
+		return nil, err
+	}
+	log.Infow("new b2 store succeeds", "bucket", bucket)
```
Comment (sysvm) [inline store/piecestore/storage/b2.go:35]: fix as `succeed to new b2 store`

**Comment by sysvm** [comment]
I think the name in unit test should be complete and meaningful.

File: test/e2e/piecestore/b2_test.go (hunk 4 above)
Comment (irrun) [reply test/e2e/piecestore/b2_test.go:12]: it's an example

File: store/piecestore/storage/b2.go (hunk 5 above)
Comment (irrun) [reply store/piecestore/storage/b2.go:103]: Using Named return parameters will enhance the code readability as one can know about the return parameters by just reading the function signature.

Comment (sysvm) [reply store/piecestore/storage/b2.go:103]: OK, I see. On the other hand, you'd better return what you want return, such as `return a, b, c`, this is readable.

**Review by sysvm — APPROVED** [review]
LGTM

**Review by ruojunm — APPROVED** [review]

**Pull request merged**